irs-edc-client/src/main/java/org/eclipse/tractusx/irs/edc/client/EDCCatalogFacade.java

```
@@ -88,30 +97,26 @@ public List<CatalogItem> fetchCatalogItems(final CatalogRequest catalogRequest)
     }
-    
+
     private static List<CatalogItem> mapToCatalogItems(final Catalog catalog) {
         if (catalog.getDatasets() == null) {
             return List.of();
         } else {
-
-            return catalog.getDatasets().stream().map(contractOffer -> {
-                final Map.Entry<String, Policy> offer = contractOffer.getOffers()
-                                                                     .entrySet()
-                                                                     .stream()
-                                                                     .findFirst()
-                                                                     .orElseThrow();
-                final var catalogItem = CatalogItem.builder()
-                                                   .itemId(contractOffer.getId())
-                                                   .assetPropId(contractOffer.getProperty(NAMESPACE_EDC_ID).toString())
-                                                   .connectorId(catalog.getId())
-                                                   .offerId(offer.getKey())
-                                                   .policy(offer.getValue());
-                if (catalog.getProperties().containsKey(JsonLdConfiguration.NAMESPACE_EDC_PARTICIPANT_ID)) {
-                    catalogItem.connectorId(
-                            catalog.getProperties().get(JsonLdConfiguration.NAMESPACE_EDC_PARTICIPANT_ID).toString());
-                }
-
-                return catalogItem.build();
+            return catalog.getDatasets().stream().map(dataset -> {
+                final Map.Entry<String, Policy> offer = dataset.getOffers()
+                                                               .entrySet()
+                                                               .stream()
+                                                               .findFirst()
+                                                               .orElseThrow();
```

maybe throw an exception with a message that explains what went wrong from a business point of view